docs/src/rules/no-func-assign.md

```diff
@@ -1,6 +1,8 @@
 ---
 title: no-func-assign
 rule_type: problem
+handled_by_typescript:
+- 2539
```

Comment:
I get `2630` in TS 5.2.2:

```
Cannot assign to 'foo' because it is a function.(2630)
```

https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAIwIYCcAUBKRBvAXwCgjg45EBeFDAbhNElgUTLh32NPKpvXqA

`2539` is the more general _"not a variable"_ error:
https://github.com/microsoft/TypeScript/blob/cd391b066b49cead0d1b1ec089ab4f4de59a16d1/src/compiler/diagnosticMessages.json#L2563-L2566

Reply:
ok, i'll update these, but can you update the upstream source too? e.g. the comments here: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/configs/eslint-recommended.ts

Reply:
https://github.com/typescript-eslint/typescript-eslint/issues/7582